src/uucore/src/lib/features/buf_copy.rs

```diff
@@ -9,27 +9,55 @@
 //! used by utilities to work around the limitations of Rust's `fs::copy` which
 //! does not handle copying special files (e.g pipes, character/block devices).
 
-use crate::error::{UError, UResult};
-use nix::unistd;
+use crate::error::UError;
+use std::io::{Read, Write};
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
it is starting to look messy with all the cfg
maybe the .rs file per platform ?
for example:
src/uu/tail/src/platform/